### 81."public static" fields should be constant

公共静态修饰的字段应该是常量

There is no good reason to declare a field "public" and "static" without also declaring it "final". Most of the time this is a kludge to share a state among several objects. But with this approach, any object can do whatever it wants with the shared state, such as setting it to`null`.

没有充分的理由就将字段声明为“ public”和“ static”而不将其声明为“ final”。 在大多数情况下，这是在多个对象之间共享状态的一种拼凑的使用方式。 但是使用这种方法，任何对象都可以在共享状态下做任何想做的事情，例如将其设置为null。

**Noncompliant Code Example**

**错误的代码示例**

```
public class Greeter {
  public static Foo foo = new Foo();
  ...
}
```

**Compliant Solution**

**正确的代码示例**


```
public class Greeter {
  public static final Foo FOO = new Foo();
  ...
}
```

