### 83."read" and "readLine" return values should be used

“read”和“readLine”返回值应该被使用

When a method is called that returns data read from some data source, that data should be stored rather than thrown away. Any other course of action is surely a bug.

This rule raises an issue when the return value of any of the following is ignored or merely null-checked:`BufferedReader.readLine()`,`Reader.read()`, and these methods in any child classes.


当调用返回从某些数据源读取的数据的方法时，应存储该数据而不是将其丢弃。 当然，任何其他措施都是错误。

当以下任何一项的返回值被忽略或仅进行空检查时，此规则会引起问题：BufferedReader.readLine() ，Reader.read() 以及任何子类中的这些方法。


**Noncompliant Code Example**

**错误的代码示例**

```
public void doSomethingWithFile(String fileName) {
  BufferedReader buffReader = null;
  try {
    buffReader = new BufferedReader(new FileReader(fileName));
    while (buffReader.readLine() != null) { // Noncompliant
      // ...
    }
  } catch (IOException e) {
    // ...
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
public void doSomethingWithFile(String fileName) {
  BufferedReader buffReader = null;
  try {
    buffReader = new BufferedReader(new FileReader(fileName));
    String line = null;
    while ((line = buffReader.readLine()) != null) {
      // ...
    }
  } catch (IOException e) {
    // ...
  }
}
```