### 85."readObject" should not be "synchronized"

readObject 方法不应该被同步修饰符（synchronized）修饰

A`readObject`method is written when a`Serializable`object needs special handling to be rehydrated from file. It should be the case that the object being created by`readObject`is only visible to the thread that invoked the method, and the`synchronized`keyword is not needed, and using`synchronized`anyway is just confusing. If this is not the case, the method should be refactored to make it the case.


**Noncompliant Code Example**

**错误的代码示例**

```
private synchronized void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Noncompliant
  //...
}
```

**Compliant Solution**

**正确的代码示例**


```
private void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Compliant
  //...
}
```