### 86."readResolve" methods should be inheritable

“readResolve”方法应该是可继承的，因为readResolve（）允许在反序列化期间对于对象的状态进行微调

The`readResolve()`method allows final tweaks to the state of an object during deserialization. Non-final classes which implement`readResolve()`, should not set its visibility to`private`since it will then be unavailable to child classes.


**Noncompliant Code Example**

**错误的代码示例**

```
public class Fruit implements Serializable {
  private static final long serialVersionUID = 1;

  private Object readResolve() throws ObjectStreamException  // Noncompliant
  {...}

  //...
}

public class Raspberry extends Fruit implements Serializable {  // No access to parent's readResolve() method
  //...
}
```

**Compliant Solution**

**正确的代码示例**


```
public class Fruit implements Serializable {
  private static final long serialVersionUID = 1;

  protected Object readResolve() throws ObjectStreamException
  {...}

  //...
}

public class Raspberry extends Fruit implements Serializable {
  //...
}
```