### 87."ResultSet.isLast()" should not be used

不应该使用“ResultSet.isLast()”方法，因为.isLast()方法会将指针的指向移动到下一个元素上，从而可能导致空指针，而.next()方法则是先判断下一个元素是否存在

There are several reasons to avoid`ResultSet.isLast()`. First, support for this method is optional for`TYPE_FORWARD_ONLY`result sets. Second, it can be expensive (the driver may need to fetch the next row to answer the question). Finally, the specification is not clear on what should be returned when the`ResultSet`is empty, so some drivers may return the opposite of what is expected.


**Noncompliant Code Example**

**错误的代码示例**

```
stmt.executeQuery("SELECT name, address FROM PERSON");
ResultSet rs = stmt.getResultSet();
while (! rs.isLast()) { // Noncompliant
  // process row
}
```

**Compliant Solution**

**正确的代码示例**


```
ResultSet rs = stmt.executeQuery("SELECT name, address FROM PERSON");
while (! rs.next()) {
  // process row
}
```