### 88."runFinalizersOnExit" should not be called

不应该调用“runFinalizersOnExit”，因为runFinalizersOnExit()方法可能导致正在被其他进程调用的对象被终止，是不安全的

Running finalizers on JVM exit is disabled by default. It can be enabled with`System.runFinalizersOnExit`and`Runtime.runFinalizersOnExit`, but both methods are deprecated because they are are inherently unsafe.

According to the Oracle Javadoc:

> It may result in finalizers being called on live objects while other threads are concurrently manipulating those objects, resulting in erratic behavior or deadlock.

If you really want to be execute something when the virtual machine begins its shutdown sequence, you should attach a shutdown hook.


**Noncompliant Code Example**

**错误的代码示例**

```
public static void main(String [] args) {
  ...
  System.runFinalizersOnExit(true);  // Noncompliant
  ...
}

protected void finalize(){
  doSomething();
}
```

**Compliant Solution**

**正确的代码示例**


```
public static void main(String [] args) {
  Runtime.addShutdownHook(new Runnable() {
    public void run(){
      doSomething();
    }
  });
  //...
```