### 31."Double.longBitsToDouble" should not be used for "int"

int类型的数据转换为lone double 会存在精度问题，不应该这样使用

`Double.longBitsToDouble`expects a 64-bit,`long`argument. Pass it a smaller value, such as an`int`and the mathematical conversion into a`double`simply won't work as anticipated because the layout of the bits will be interpreted incorrectly, as if a child were trying to use an adult's gloves.


**Noncompliant Code Example**

**错误的代码示例**

```
int i = 42;
double d = Double.longBitsToDouble(i);  // Noncompliant
```


