
### 37."equals" methods should be symmetric and work for subclasses

equals(Object obj)和hashCode()应该成对出现


According to the Java Language Specification, there is a contract between`equals(Object)`and`hashCode()`:

> If two objects are equal according to the`equals(Object)`method, then calling the`hashCode`method on each of the two objects must produce the same integer result.
> 
> It is not required that if two objects are unequal according to the`equals(java.lang.Object)`method, then calling the`hashCode`method on each of the two objects must produce distinct integer results.
> 
> However, the programmer should be aware that producing distinct integer results for unequal objects may improve the performance of hashtables.

In order to comply with this contract, those methods should be either both inherited, or both overridden.

**Noncompliant Code Example**

**错误的代码示例**

```
class MyClass {    // Noncompliant - should also override "hashCode()"

  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

}
```


**Compliant Solution**

**正确的代码示例**

```
lass MyClass {    // Compliant

  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

  @Override
  public int hashCode() {
    /* ... */
  }

}
```


