
### 39."equals(Object obj)" should test argument type

因为equals方法能接受泛型参数，所以需要对于传递的参数类型进行验证


Because the`equals`method takes a generic`Object`as a parameter, any type of object may be passed to it. The method should not assume it will only be used to test objects of its class type. It must instead check the parameter's type.

**Noncompliant Code Example**

**错误的代码示例**

```
public boolean equals(Object obj) {
  MyClass mc = (MyClass)obj;  // Noncompliant
  // ...
}
```


**Compliant Solution**

**正确的代码示例**

```
public boolean equals(Object obj) {
  if (obj == null)
    return false;

  if (this.getClass() != obj.getClass())
    return false;

  MyClass mc = (MyClass)obj;
  // ...
}
```


