
### 40."Exception" should not be caught when not required by called methods

当调用的方法不需要捕获异常时，不应该捕获到异常（主要针对于运行时异常和检查异常）

运行时如果异常应该能正常捕获到，并提醒开发人员调整

Catching`Exception`seems like an efficient way to handle multiple possible exceptions. Unfortunately, it traps all exception types, both checked and runtime exceptions, thereby casting too broad a net. Indeed, was it really the intention of developers to also catch runtime exceptions? To prevent any misunderstanding, if both checked and runtime exceptions are really expected to be caught, they should be explicitly listed in the`catch`clause.

This rule raises an issue if`Exception`is caught when it is not explicitly thrown by a method in the`try`block.

**Noncompliant Code Example**

**错误的代码示例**

```
try {
  // do something that might throw an UnsupportedDataTypeException or UnsupportedEncodingException
} catch (Exception e) { // Noncompliant
  // log exception ...
}
```


**Compliant Solution**

**正确的代码示例**

```
try {
  // do something
} catch (UnsupportedEncodingException|UnsupportedDataTypeException|RuntimeException e) {
  // log exception ...
}
```
或者

```
try {
  // do something
} catch (UnsupportedEncodingException|UnsupportedDataTypeException e) {
  // log exception ...
}
```


