### 71."NullPointerException" should not be caught

“NullPointerException”不应被捕获

`NullPointerException`should be avoided, not caught. Any situation in which`NullPointerException`is explicitly caught can easily be converted to a`null`test, and any behavior being carried out in the catch block can easily be moved to the "is null" branch of the conditional.

NullPointerException应该避免，而不是被捕获。 在任何情况下，显示捕获NullPointerException都可以轻松转换为null测试，并且可以轻松将在catch块中执行的任何行为移至条件语句的“ is null”分支。

**Noncompliant Code Example**

**错误的代码示例**

```
public int lengthPlus(String str) {
  int len = 2;
  try {
    len += str.length();
  }
  catch (NullPointerException e) {
    log.info("argument was null");
  }
  return len;
}
```

**Compliant Solution**

**正确的代码示例**


```
public int lengthPlus(String str) {
  int len = 2;

  if (str != null) {
    len += str.length();
  }
  else {
    log.info("argument was null");
  }
  return len;
}
```