### 73."Object.finalize()" should remain protected (versus public) when overriding

Object.finalize() 方法应该只有垃圾回收器才可以调用

The contract of the`Object.finalize()`method is clear: only the Garbage Collector is supposed to call this method.

Making this method public is misleading, because it implies that any caller can use it.


**Noncompliant Code Example**

**错误的代码示例**

```
public class MyClass {

  @Override
  public void finalize() {    // Noncompliant
    /* ... */
  }
}
```
