### 74."Object.wait(...)" and "Condition.await(...)" should be called inside a "while" loop

Object.wait(…)”和“condition .wait(…)”应该在while循环中调用

According to the documentation of the Java`Condition`interface:

> When waiting upon a`Condition`, a "spurious wakeup" is permitted to occur, in general, as a concession to the underlying platform semantics. This has little practical impact on most application programs as a Condition should always be waited upon in a loop, testing the state predicate that is being waited for. An implementation is free to remove the possibility of spurious wakeups but it is recommended that applications programmers always assume that they can occur and so always wait in a loop.

The same advice is also found for the`Object.wait(...)`method:

> waits should always occur in loops, like this one:
> 
> synchronized (obj) {
>   while (<condition does not hold>){
>     obj.wait(timeout);
>   }
>    ... // Perform action appropriate to condition
> }


**Noncompliant Code Example**

**错误的代码示例**

```
synchronized (obj) {
  if (!suitableCondition()){
    obj.wait(timeout);   //the thread can wake up even if the condition is still false
  }
   ... // Perform action appropriate to condition
}
```

**Compliant Solution**

**正确的代码示例**


```
synchronized (obj) {
  while (!suitableCondition()){
    obj.wait(timeout);
  }
   ... // Perform action appropriate to condition
}
```