### 75."Object.wait(...)" should never be called on objects that implement "java.util.concurrent.locks.Condition"

实现“java.util.concurrent.locks.Condition”的对象不应该调用“Object.wait(…)”方法

所在的超类不同使用场景也不同，wait一般用于Synchronized中，而await只能用于ReentrantLock锁中

From the Java API documentation:

> `Condition`factors out the`Object`monitor methods (`wait`,`notify`and`notifyAll`) into distinct objects to give the effect of having multiple wait-sets per object, by combining them with the use of arbitrary Lock implementations. Where a`Lock`replaces the use of`synchronized`methods and statements, a`Condition`replaces the use of the`Object`monitor methods.

The purpose of implementing the`Condition`interface is to gain access to its more nuanced`await`methods. Therefore, calling the method`Object.wait(...)`on a class implementing the`Condition`interface is silly and confusing.


**Noncompliant Code Example**

**错误的代码示例**

```
final Lock lock = new ReentrantLock();
final Condition notFull  = lock.newCondition();
...
notFull.wait();
```

**Compliant Solution**

**正确的代码示例**


```
final Lock lock = new ReentrantLock();
final Condition notFull  = lock.newCondition();
...
notFull.await();
```

