### 79."private" methods called only by inner classes should be moved to those classes

只由内部类调用的“私有”方法应该放到这些类里


When a`private`method is only invoked by an inner class, there's no reason not to move it into that class. It will still have the same access to the outer class' members, but the outer class will be clearer and less cluttered.


**Noncompliant Code Example**

**错误的代码示例**

```
public class Outie {
  private int i=0;

  private void increment() {  // Noncompliant
    i++;
  }

  public class Innie {
    public void doTheThing() {
      Outie.this.increment();
    }
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
public class Outie {
  private int i=0;

  public class Innie {
    public void doTheThing() {
      Outie.this.increment();
    }

    private void increment() {
      Outie.this.i++;
    }
  }
}
```