
### 15."Arrays.stream" should be used for primitive arrays

流应该被用在基本类型数组上.

For arrays of objects,`Arrays.asList(T ... a).stream()`and`Arrays.stream(array)`are basically equivalent in terms of performance. However, for arrays of primitives, using`Arrays.asList`will force the construction of a list of boxed types, and then use_that_list as a stream. On the other hand,`Arrays.stream`uses the appropriate primitive stream type (`IntStream`,`LongStream`,`DoubleStream`) when applicable, with much better performance.



**Noncompliant Code Example**

**错误的代码示例**

```
Arrays.asList("a1", "a2", "b1", "c2", "c1").stream()
    .filter(...)
    .forEach(...);

Arrays.asList(1, 2, 3, 4).stream() // Noncompliant
    .filter(...)
    .forEach(...);
```


**Compliant Solution**

**正确的代码示例**

```
Arrays.asList("a1", "a2", "b1", "c2", "c1").stream()
    .filter(...)
    .forEach(...);

int[] intArray = new int[]{1, 2, 3, 4};
Arrays.stream(intArray)
    .filter(...)
    .forEach(...);
```

