### 101."StringBuilder" and "StringBuffer" should not be instantiated with a character

"StringBuilder" 和 "StringBuffer"不应该被单个字符实例化

Instantiating a`StringBuilder`or a`StringBuffer`with a character is misleading because most Java developers would expect the character to be the initial value of the`StringBuffer`.

What actually happens is that the int representation of the character is used to determine the initial size of the`StringBuffer`.

实际发生的是，字符的int值表示用于确定StringBuffer的初始大小。


**Noncompliant Code Example**

**错误的代码示例**

```
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
```

**Compliant Solution**

**正确的代码示例**


```
StringBuffer foo = new StringBuffer("x");
```