### 103."switch case" clauses should not have too many lines of code


“switch case”子句不应包含太多代码行


The`switch`statement should be used only to clearly define some new branches in the control flow. As soon as a`case`clause contains too many statements this highly decreases the readability of the overall control flow statement. In such case, the content of the`case`clause should be extracted into a dedicated method.

switch语句仅应用于明确定义控制流中的一些新分支。 一旦case子句包含太多语句，就会大大降低整个控制流语句的可读性。 在这种情况下，case子句的内容应提取到专用方法中。

**Noncompliant Code Example**

**错误的代码示例**

```
switch (myVariable) {
  case 0: // Noncompliant: 6 lines till next case
    methodCall1("");
    methodCall2("");
    methodCall3("");
    methodCall4("");
    break;
  case 1:
  ...
}
```

**Compliant Solution**

**正确的代码示例**


```
switch (myVariable) {
  case 0:
    doSomething()
    break;
  case 1:
  ...
}
...
private void doSomething(){
    methodCall1("");
    methodCall2("");
    methodCall3("");
    methodCall4("");
}
```
