### 104."switch" statements should have "default" clauses

“switch”语句应具有“default”子句，控制除了case以外的其他场景

The requirement for a final`default`clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken.

最终默认子句的要求是防御性编程。该子句应采取适当的措施，或包含有关为何不采取任何措施的适当注释。


**Noncompliant Code Example**

**错误的代码示例**

```
switch (param) {  //missing default clause
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
}

switch (param) {
  default: // default clause should be the last one
    error();
    break;
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
}
```

**Compliant Solution**

**正确的代码示例**


```
switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}
```

**Exceptions**

**例外**

If the`switch`parameter is an`Enum`and if all the constants of this enum are used in the`case`statements, then no`default`clause is expected.

如果switch参数是一个Enum，并且在case语句中使用了该枚举的所有常量，则不应使用default子句。

Example:

```
public enum Day {
    SUNDAY, MONDAY
}
...
switch(day) {
  case SUNDAY:
    doSomething();
    break;
  case MONDAY:
    doSomethingElse();
    break;
}
```
