### 105."switch" statements should have at least 3 "case" clauses

“switch”语句应至少包含3个“ case”子句，case子句<=2的场景，用if...else即可

`switch`statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with`if`statements.

当根据同一表达式的值有许多不同的情况时，switch语句很有用。

但是，仅在一种或两种情况下，使用if语句将使代码更具可读性。


**Noncompliant Code Example**

**错误的代码示例**

```
switch (variable) {
  case 0:
    doSomething();
    break;
  default:
    doSomethingElse();
    break;
}
```

**Compliant Solution**

**正确的代码示例**


```
if (variable == 0) {
  doSomething();
} else {
  doSomethingElse();
}
```