### 106."switch" statements should not be nested

“switch”语句不应嵌套

Nested`switch`structures are difficult to understand because you can easily confuse the cases of an inner`switch`as belonging to an outer statement. Therefore nested`switch`statements should be avoided.

嵌套开关结构很难理解，因为您很容易将内部开关的情况混淆为属于外部语句的情况。因此，应避免嵌套的switch语句。

Specifically, you should structure your code to avoid the need for nested`switch`statements, but if you cannot, then consider moving the inner`switch`to another function.

具体来说，您应该对代码进行结构化，以避免需要嵌套的switch语句，但是，如果不能，则考虑将内部switch移至另一个函数。


**Noncompliant Code Example**

**错误的代码示例**

```
void foo(int n, int m) {
  switch (n) {
    case 0:
      switch (m) {  // Noncompliant; nested switch
        // ...
      }
    case 1:
      // ...
    default:
      // ...
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
void foo(int n, int m) {
  switch (n) {
    case 0:
      bar(m);
    case 1:
      // ...
    default:
      // ...
  }
}

void bar(int m){
  switch(m) {
    // ...
  }
}
```