### 109."this" should not be exposed from constructors

“this”不应从构造函数中暴露

n single-threaded environments, the use of`this`in constructors is normal, and expected. But in multi-threaded environments, it could expose partially-constructed objects to other threads, and should be used with caution.

在单线程环境中，在构造函数中使用this是正常的，并且是预期的。但是在多线程环境中，它可能会将部分构造的对象暴露给其他线程，因此应谨慎使用。

The classic example is a class with a`static`list of its instances. If the constructor stores`this`in the list, another thread could access the object before it's fully-formed. Even when the storage of`this`is the last instruction in the constructor, there's still a danger if the class is not`final`. In that case, the initialization of subclasses won't be complete before`this`is exposed.

示例是带有实例静态列表的类。 如果构造函数将其存储在列表中，则另一个线程可以在对象完全形成之前访问该对象。 即使将其存储为构造函数中的最后一条指令，如果类不是最终的，仍然存在危险。 在这种情况下，子类的初始化在公开之前是不会完成的。

This rule raises an issue when`this`is assigned to any globally-visible object in a constructor, and when it is passed to the method of another object in a constructor

当将此规则分配给构造函数中的任何全局可见对象时，以及将其传递给构造函数中另一个对象的方法时，此规则都会引起问题


**Noncompliant Code Example**

**错误的代码示例**

```
public class Monument {

  public static final List<Monument> ALL_MONUMENTS = new ArrayList()<>;
  // ...

  public Monument(String location, ...) {
    ALL_MONUMENTS.add(this);  // Noncompliant; passed to a method of another object

    this.location = location;
    // ...
  }
}
```

**Exceptions**

**例外**

This rule ignores instances of assigning`this`directly to a`static`field of the same class because that case is covered by S3010.

该规则忽略将其直接分配给同一类的静态字段的实例，因为这种情况已在S3010中涵盖。