### 110."Thread.run()" should not be called directly

不应直接调用“Thread.run()”

The purpose of the`Thread.run()`method is to execute code in a separate, dedicated thread. Calling this method directly doesn't make sense because it causes its code to be executed in the current thread.

Thread和Runnable里面的run方法设计的目的是让run方法里面的代码在不同的线程中执行。。直接调用此方法没有意义，因为它导致其代码在当前线程中执行。

To get the expected behavior, call the`Thread.start()`method instead.

为了获得预期的行为，请改为调用Thread.start()方法。


**Noncompliant Code Example**

**错误的代码示例**

```
Thread myThread = new Thread(runnable);
myThread.run(); // Noncompliant
```

**Compliant Solution**

**正确的代码示例**


```
Thread myThread = new Thread(runnable);
myThread.start(); // Compliant
```