### 21."close()" calls should not be redundant

不应该有多余的close()方法的调用

Java 7's try-with-resources structure automatically handles closing the resources that the`try`itself opens. Thus, adding an explicit`close()`call is redundant and potentially confusing.


**Noncompliant Code Example**

**错误的代码示例**

```
class Team implements Cloneable {  // Noncompliant
  private Person coach;
  private List<Person> players;
  public void addPlayer(Person p) {...}
  public Person getCoach() {...}
}
```


**Compliant Solution**

**正确的代码示例**

```
class Team implements Cloneable {
  private Person coach;
  private List<Person> players;
  public void addPlayer(Person p) { ... }
  public Person getCoach() { ... }

  @Override
  public Object clone() {
    Team clone = (Team) super.clone();
    //...
  }
}
```


