### 22."collect" should be used with "Streams" instead of "list::add"

使用 Streams 时，应该用 collect，因为collect 的进程是并行且安全的

While you can use either`forEach(list::add)`or`collect`with a`Stream`,`collect`is by far the better choice because it's automatically thread-safe and parallellizable.


**Noncompliant Code Example**

**错误的代码示例**

```
List<String> bookNames = new ArrayList<>();
books.stream().filter(book -> book.getIsbn().startsWith("0"))
                .map(Book::getTitle)
                .forEach(bookNames::add);  // Noncompliant
```


**Compliant Solution**

**正确的代码示例**

```
List<String> bookNames = books.stream().filter(book -> book.getIsbn().startsWith("0"))
                .map(Book::getTitle)
                .collect(Collectors.toList());
```


