### 24."compareTo" results should not be checked for specific values

“compareTo”结果不应检查特定值，通常检查`>0`或者`<0`

While most`compareTo`methods return -1, 0, or 1, some do not, and testing the result of a`compareTo`against a specific value other than 0 could result in false negatives.


**Noncompliant Code Example**

**错误的代码示例**

```
if (myClass.compareTo(arg) == -1) {  // Noncompliant
  // ...
}
```


**Compliant Solution**

**正确的代码示例**

```
if (myClass.compareTo(arg) < 0) {
  // ...
}
```


