### 26."compareTo" should not be overloaded

compareTo 的返回值不能是Integer.MIN_VALUE

It is the sign, rather than the magnitude of the value returned from`compareTo`that matters. Returning`Integer.MIN_VALUE`does_not_convey a higher degree of inequality, and doing so can cause errors because the return value of`compareTo`is sometimes inversed, with the expectation that negative values become positive. However, inversing`Integer.MIN_VALUE`yields`Integer.MIN_VALUE`rather than`Integer.MAX_VALUE`.


**Noncompliant Code Example**

**错误的代码示例**

```
public int compareTo(MyClass) {
  if (condition) {
    return Integer.MIN_VALUE;  // Noncompliant
  }
```


**Compliant Solution**

**正确的代码示例**

```
public int compareTo(MyClass) {
  if (condition) {
    return -1;
  }
```


