### 27."DateUtils.truncate" from Apache Commons Lang library should not be used

Apache Commons Lang 的日期截取方法 DateUtils.truncate() 不会四舍五入，不应该被使用

The use of the`ZonedDateTime`class introduced in Java 8 to truncate a date can be significantly faster than the`DateUtils`class from Commons Lang.

**Note**that this rule is automatically disabled when the project's`sonar.java.source`is lower than`8`.


**Noncompliant Code Example**

**错误的代码示例**

```
public Date trunc(Date date) {
  return DateUtils.truncate(date, Calendar.SECOND);  // Noncompliant
}
```


**Compliant Solution**

**正确的代码示例**

```
public Date trunc(Date date) {
  Instant instant = date.toInstant();
  ZonedDateTime zonedDateTime = instant.atZone(ZoneId.systemDefault());
  ZonedDateTime truncatedZonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.SECONDS);
  Instant truncatedInstant = truncatedZonedDateTime.toInstant();
  return Date.from(truncatedInstant);
}
```


