### 30."deleteOnExit" should not be used

java程序执行结束后，有自己的回收机制，不需要用deleteOnExit回收临时文件

Use of`File.deleteOnExit()`is not recommended for the following reasons:

-   The deletion occurs only in the case of a normal JVM shutdown but not when the JVM crashes or is killed.
-   For each file handler, the memory associated with the handler is released only at the end of the process.


**Noncompliant Code Example**

**错误的代码示例**

```
File file = new File("file.txt");
file.deleteOnExit();  // Noncompliant
```

