### 60."Iterator.hasNext()" should not call "Iterator.next()"

Iterator.hasNext()是判断是否有下一个元素，但并不会改变指针指向，而next（）会改变指针指向，如果没有下一个元素，会直接报空指针异常

Calling`Iterator.hasNext()`is not supposed to have any side effects, and therefore should not change the state of the iterator.`Iterator.next()`advances the iterator by one item. So calling it inside`Iterator.hasNext()`, breaks the`hasNext()`contract, and will lead to unexpected behavior in production.


**Noncompliant Code Example**

**错误的代码示例**

```
public class FibonacciIterator implements Iterator<Integer>{
...
@Override
public boolean hasNext() {
  if(next() != null) {
    return true;
  }
  return false;
}
...
}
```

