### 111."Thread.sleep" should not be used in tests

Thread.sleep()方法不应该在测试过程中被调用，建议写根据具体条件执行的异步语句。

Using`Thread.sleep`in a test is just generally a bad idea. It creates brittle tests that can fail unpredictably depending on environment ("Passes on my machine!") or load. Don't rely on timing (use mocks) or use libraries such as`Awaitility`for asynchroneous testing.

在测试中使用Thread.sleep通常不是一个好主意。它创建的测试是脆弱的，可能会因环境（“在我的机器上通过了！”）或负载而意外失败。不要依赖定时器（使用模拟）或使用诸如Awaitility之类的库进行异步测试。


**Noncompliant Code Example**

**错误的代码示例**

```
@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  Thread.sleep(500);  // Noncompliant
  // assertions...
}
```

**Compliant Solution**

**正确的代码示例**


```
@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  await().atMost(2, Duration.SECONDS).until(didTheThing());  // Compliant
  // assertions...
}

private Callable<Boolean> didTheThing() {
  return new Callable<Boolean>() {
    public Boolean call() throws Exception {
      // check the condition that must be fulfilled...
    }
  };
}
```