### 112."ThreadGroup" should not be used

不应使用“ThreadGroup”

There is little valid reason to use the methods of the`ThreadGroup`class. Some are deprecated (`allowThreadSuspension()`,`resume()`,`stop()`, and`suspend()`), some are obsolete, others aren't thread-safe, and still others are insecure (`activeCount()`,`enumerate()`) . For these reasons, any use of`ThreadGroup`is suspicious and should be avoided.

没有什么充分的理由使用ThreadGroup类的方法。有些已弃用(allowThreadSuspension(),resume(),stop(), andsuspend()),，有些已经过时，有些不是线程安全的，还有一些是不安全的 (activeCount(),enumerate()) 。由于这些原因，对ThreadGroup的任何使用都是可疑的，应避免使用。



**Noncompliant Code Example**

**错误的代码示例**

```
ThreadFactory threadFactory = Executors.defaultThreadFactory();
ThreadPoolExecutor executorPool = new ThreadPoolExecutor(3, 10, 5, TimeUnit.SECONDS, new ArrayBlockingQueue<Runnable>(2), threadFactory);

for (int i = 0; i < 10; i++) {
  executorPool.execute(new JobThread("Job: " + i));
}

System.out.println(executorPool.getActiveCount()); // Compliant
executorPool.shutdown();
```
