### 113."ThreadLocal.withInitial" should be preferred

应优先考虑ThreadLocal.withInitial()方法

Java 8 introduced`ThreadLocal.withInitial`which is a simpler alternative to creating an anonymous inner class to initialise a`ThreadLocal`instance.

Java 8引入了ThreadLocal.withInitial，它是创建匿名内部类以初始化ThreadLocal实例的一种更简单的选择。

This rule raises an issue when a`ThreadLocal`anonymous inner class can be replaced by a call to`ThreadLocal.withInitial`.

当可以将ThreadLocal匿名内部类替换为对ThreadLocal.withInitial的调用时，此规则会引起问题。


**Noncompliant Code Example**

**错误的代码示例**

```
ThreadLocal<List<String>> myThreadLocal =
    new ThreadLocal<List<String>>() { // Noncompliant
        @Override
        protected List<String> initialValue() {
            return new ArrayList<String>();
        }
    };
```

**Compliant Solution**

**正确的代码示例**


```
ThreadLocal<List<String>> myThreadLocal = ThreadLocal.withInitial(ArrayList::new);
```

