### 114."Threads" should not be used where "Runnables" are expected

不应在需要“Runnables”的地方使用“Threads”

While it is technically correct to use a`Thread`where a`Runnable`is called for, the semantics of the two objects are different, and mixing them is a bad practice that will likely lead to headaches in the future.

虽然在需要Runnable的地方使用Thread在技术上是正确的，但是两个对象的语义是不同的，并且将它们混合使用是一种不好的做法，将来可能会引起麻烦。

The crux of the issue is that`Thread`is a larger concept than`Runnable`. A`Runnable`is an object whose running should be managed. A`Thread`expects to manage the running of itself or other`Runnables`.

问题的症结在于，线程Thread是一个比Runnable更大的概念。可运行对象是应管理其运行的对象。线程希望管理其自身或其他Runnable的运行。


**Noncompliant Code Example**

**错误的代码示例**

```
public static void main(String[] args) {
		Thread r =new Thread() {
			int p;
			@Override
			public void run() {
				while(true)
					System.out.println("a");
			}
		};
		new Thread(r).start();  // Noncompliant
```

**Compliant Solution**

**正确的代码示例**


```
public static void main(String[] args) {
		Runnable r =new Runnable() {
			int p;
			@Override
			public void run() {
				while(true)
					System.out.println("a");
			}
		};
		new Thread(r).start();
```