### 116."toArray" should be passed an array of the proper type

“toArray”应传递适当类型的数组

Given no arguments, the`Collections.toArray`method returns an`Object []`, which will cause a`ClassCastException`at runtime if you try to cast it to an array of the proper class. Instead, pass an array of the correct type in to the call.

如果没有给出任何参数，Collections.toArray方法将返回一个Object []，如果您尝试将其强制转换为适当类的数组，则它将在运行时导致ClassCastException。
相反，因该将正确类型的数组传递给调用。


**Noncompliant Code Example**

**错误的代码示例**

```
public String [] getStringArray(List<String> strings) {
  return (String []) strings.toArray();  // Noncompliant; ClassCastException thrown
}
```

**Compliant Solution**

**正确的代码示例**


```
public String [] getStringArray(List<String> strings) {
  return strings.toArray(new String[0]);
}
```

