### 117."toString()" and "clone()" methods should not return null

“toString()”和“clone()”方法不应返回null

Calling`toString()`or`clone()`on an object should always return a string or an object. Returning`null`instead contravenes the method's implicit contract.

在对象上调用toString()或clone()应该始终返回字符串或对象。相反，返回null会违反该方法的隐式协定。

**Noncompliant Code Example**

**错误的代码示例**

```
public String toString () {
  if (this.collection.isEmpty()) {
    return null; // Noncompliant
  } else {
    // ...
 {code}
```

**Compliant Solution**

**正确的代码示例**


```
public String toString () {
  if (this.collection.isEmpty()) {
    return "";
  } else {
    // ...
```