### 118."toString()" should never be called on a String object

永远不要在String对象上调用“toString()”

Invoking a method designed to return a string representation of an object which is already a string is a waste of keystrokes. This redundant construction may be optimized by the compiler, but will be confusing in the meantime.

调用旨在返回已经是字符串的对象的字符串表示形式的方法是浪费按键。编译器可能会优化此冗余结构，但与此同时会造成混淆。


**Noncompliant Code Example**

**错误的代码示例**

```
String message = "hello world";
System.out.println(message.toString()); // Noncompliant;
```

**Compliant Solution**

**正确的代码示例**


```
String message = "hello world";
System.out.println(message);
```