### 121."wait" should not be called when multiple locks are held

当持有多个锁时，不应调用“wait”


When two locks are held simultaneously, a`wait`call only releases one of them. The other will be held until some other thread requests a lock on the awaited object. If no unrelated code tries to lock on that object, then all other threads will be locked out, resulting in a deadlock.

当同时持有两个锁时，等待调用只会释放其中一个。另一个将一直保留，直到其他一些线程请求对等待的对象进行锁定。如果没有不相关的代码尝试锁定该对象，则所有其他线程将被锁定，从而导致死锁。


**Noncompliant Code Example**

**错误的代码示例**

```
synchronized (this.mon1) {  // threadB can't enter this block to request this.mon2 lock & release threadA
	synchronized (this.mon2) {
		this.mon2.wait();  // Noncompliant; threadA is stuck here holding lock on this.mon1
	}
}
```
