### 122."wait", "notify" and "notifyAll" should only be called when a lock is obviously held on an object

“wait”、“notify”和“notifyAll”只应在对象上明显持有锁时（如被synchronized修饰）调用

By contract, the method`Object.wait(...)`,`Object.notify()`and`Object.notifyAll()`should be called by a thread that is the owner of the object's monitor. If this is not the case an`IllegalMonitorStateException`exception is thrown. This rule reinforces this constraint by making it mandatory to call one of these methods only inside a`synchronized`method or statement.

根据合同，应由作为对象监视器的所有者的线程调用Object.wait(...)，Object.notify()和Object.notifyAll()方法。如果不是这种情况，则抛出IllegalMonitorStateException异常。该规则通过强制仅在synchronized方法或语句内调用这些方法之一来加强此约束。


**Noncompliant Code Example**

**错误的代码示例**

```
private void removeElement() {
  while (!suitableCondition()){
    obj.wait();
  }
  ... // Perform removal
}
```
或者

```
private void removeElement() {
  while (!suitableCondition()){
    wait();
  }
  ... // Perform removal
}
```


**Compliant Solution**

**正确的代码示例**


```
private void removeElement() {
  synchronized(obj) {
    while (!suitableCondition()){
      obj.wait();
    }
    ... // Perform removal
  }
}
```
或者

```
private synchronized void removeElement() {
  while (!suitableCondition()){
    wait();
  }
  ... // Perform removal
}
```
