### 123."wait(...)" should be used instead of "Thread.sleep(...)" when a lock is held

持有锁时，应使用“ wait(...)”代替“ Thread.sleep(...)”

如果在当前线程持有锁时调用thread .sleep(…)，可能会导致性能和可伸缩性问题，或者更糟的情况是死锁。所以当锁被持有时，应该使用“wait(…)”而不是“Thread.sleep(…)”方法

If`Thread.sleep(...)`is called when the current thread holds a lock, it could lead to performance and scalability issues, or even worse to deadlocks because the execution of the thread holding the lock is frozen. It's better to call`wait(...)`on the monitor object to temporarily release the lock and allow other threads to run.

如果在当前线程持有锁时调用 Thread.sleep(...) ，则可能导致性能和可伸缩性问题，甚至导致死锁，因为持有锁的线程的执行被冻结。最好在监视对象上调用 wait(...) 以暂时释放锁并允许其他线程运行。


**Noncompliant Code Example**

**错误的代码示例**

```
public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      Thread.sleep(200);
    }
    process();
  }
  ...
}
```

**Compliant Solution**

**正确的代码示例**


```
public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      monitor.wait(200);
    }
    process();
  }
  ...
}
```