### 124."write(byte[],int,int)" should be overridden

当直接子类化java.io.OutputStream 或者 java.io.FilterOutputStream时，因为不能每次传递一个byte的数据，“write(byte[]，int,int)”方法应该被重写

When directly subclassing`java.io.OutputStream`or`java.io.FilterOutputStream`, the only requirement is that you implement the method`write(int)`. However most uses for such streams don't write a single byte at a time and the default implementation for`write(byte[],int,int)`will call`write(int)`for every single byte in the array which can create a lot of overhead and is utterly inefficient. It is therefore strongly recommended that subclasses provide an efficient implementation of`write(byte[],int,int)`.

直接子类化java.io.OutputStream或java.io.FilterOutputStream时，唯一的要求就是实现方法 write(int)  。但是，此类流的大多数用法不会一次写入单个字节，而 write(byte[],int,int)  的默认实现将为数组中的每个单个字节调用 write(int) ，这会创建很多字节开销，并且完全没有效率。因此，强烈建议子类提供 write(byte[],int,int) 的有效实现。

This rule raises an issue when a direct subclass of`java.io.OutputStream`or`java.io.FilterOutputStream`doesn't provide an override of`write(byte[],int,int)`.

当java.io.OutputStream或java.io.FilterOutputStream的直接子类不提供 write(byte[],int,int) 的重写时，此规则会引起问题。

**Noncompliant Code Example**

**错误的代码示例**

```
public class MyStream extends OutputStream { // Noncompliant
    private FileOutputStream fout;

    public MyStream(File file) throws IOException {
        fout = new FileOutputStream(file);
    }

    @Override
    public void write(int b) throws IOException {
        fout.write(b);
    }

    @Override
    public void close() throws IOException {
        fout.write("\n\n".getBytes());
        fout.close();
        super.close();
    }
}
```

**Compliant Solution**

**正确的代码示例**


```
public class MyStream extends OutputStream {
    private FileOutputStream fout;

    public MyStream(File file) throws IOException {
        fout = new FileOutputStream(file);
    }

    @Override
    public void write(int b) throws IOException {
        fout.write(b);
    }

    @Override
    public void write(byte[] b, int off, int len) throws IOException {
        fout.write(b, off, len);
    }

    @Override
    public void close() throws IOException {
        fout.write("\n\n".getBytes());
        fout.close();
        super.close();
    }
}
```

**Exceptions**

**例外**

This rule doesn't raise an issue when the class is declared`abstract`.

当类被声明为抽象时，此规则不会引起任何问题。