### 125."writeObject" should not be the only "synchronized" code in a class

“writeObject”不应该是类中唯一的“synchronized”修饰符修饰的方法

The purpose of synchronization is to ensure that only one thread executes a given block of code at a time. There's no real problem with marking`writeObject``synchronized`, but it's highly suspicious if this serialization-related method is the only`synchronized`code in a`class`.

同步的目的是确保一次只有一个线程执行给定的代码块。标记writeObject同步并没有真正的问题，但是如果与此序列化相关的方法是类中唯一的同步代码，那将非常可疑。


**Noncompliant Code Example**

**错误的代码示例**

```
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
```

**Compliant Solution**

**正确的代码示例**


```
StringBuffer foo = new StringBuffer("x");
```