### 126.@FunctionalInterface annotation should be used to flag Single Abstract Method interfaces

**废弃**

@FunctionalInterface批注应用于标记单个抽象方法接口

A Single Abstract Method (SAM) interface is a Java interface containing only one method. The Java API is full of SAM interfaces, such as`java.lang.Runnable`,`java.awt.event.ActionListener`,`java.util.Comparator`and`java.util.concurrent.Callable`. SAM interfaces have a special place in Java 8 because they can be implemented using Lambda expressions or Method references.

单一抽象方法（SAM）接口是仅包含一种方法的Java接口。Java API充满了SAM接口，例如java.lang.Runnable，java.awt.event.ActionListener，java.util.Comparator和java.util.concurrent.Callable。SAM接口在Java 8中占有特殊的位置，因为可以使用Lambda表达式或方法引用来实现它们。

Using`@FunctionalInterface`forces a compile break when an additional, non-overriding abstract method is added to a SAM, which would break the use of Lambda implementations.

使用@FunctionalInterface会在向SAM中添加其他不可覆盖的抽象方法时强制编译中断，这会中断Lambda实现的使用。

**Note**that this rule is automatically disabled when the project's`sonar.java.source`is lower than`8`.

请注意，当项目的sonar.java.source低于8时，将自动禁用此规则。


**Noncompliant Code Example**

**错误的代码示例**

```
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
```

**Compliant Solution**

**正确的代码示例**


```
StringBuffer foo = new StringBuffer("x");
```

**Deprecated**

**不推荐使用**

This rule is deprecated, and will eventually be removed.

此规则已弃用，最终将被删除。