### 127.A "for" loop update clause should move the counter in the right direction

“for”循环更新子句应该将计数器移动到正确的方向

A`for`loop with a counter that moves in the wrong direction is not an infinite loop. Because of wraparound, the loop will eventually reach its stop condition, but in doing so, it will run many, many more times than anticipated, potentially causing unexpected behavior.

计数器朝错误方向移动的for循环不是无限循环。由于存在环绕，因此循环最终将达到其停止条件，但这样做将使循环运行比预期的次数多很多，从而可能导致意外的行为。


**Noncompliant Code Example**

**错误的代码示例**

```
public void doSomething(String [] strings) {
  for (int i = 0; i < strings.length; i--) { // Noncompliant;
    String string = strings[i];  // ArrayIndexOutOfBoundsException when i reaches -1
    //...
  }
```

**Compliant Solution**

**正确的代码示例**


```
public void doSomething(String [] strings) {
  for (int i = 0; i < strings.length; i++) {
    String string = strings[i];
    //...
  }
```