### 128.A "while" loop should be used instead of a "for" loop

当for循环中只定义了条件表达式，并且缺少初始化和增量表达式时，应该使用while循环来增加可读性

When only the condition expression is defined in a`for`loop, and the initialization and increment expressions are missing, a`while`loop should be used instead to increase readability.

当在for循环中仅定义条件表达式且缺少初始化和增量表达式时，应改用while循环以提高可读性。


**Noncompliant Code Example**

**错误的代码示例**

```
for (;condition;) { /*...*/ }
```


**Compliant Solution**

**正确的代码示例**

```
while (condition) { /*...*/ }
```
