### 129.A close curly brace should be located at the beginning of a line

右花括号应位于行首

Shared coding conventions make it possible for a team to efficiently collaborate. This rule makes it mandatory to place a close curly brace at the beginning of a line.

共享的编码约定使团队可以有效地进行协作。此规则使必须在行的开头放置一个大括号。


**Noncompliant Code Example**

**错误的代码示例**

```
if(condition) {
  doSomething();}
```

**Compliant Solution**

**正确的代码示例**


```
if(condition) {
  doSomething();
}
```

**Exceptions**

**例外**

When blocks are inlined (open and close curly braces on the same line), no issue is triggered.

内联块时（在同一行上打开和关闭大括号），不会触发任何问题。

```
if(condition) {doSomething();}
```

