### 130.A conditionally executed single line should be denoted by indentation

有if条件执行的单行应该用缩进表示（未添加{}时）

In the absence of enclosing curly braces, the line immediately after a conditional is the one that is conditionally executed. By both convention and good practice, such lines are indented. In the absence of both curly braces and indentation the intent of the original programmer is entirely unclear and perhaps not actually what is executed. Additionally, such code is highly likely to be confusing to maintainers.

在没有花括号的情况下，紧接条件语句的行是有条件执行的行。按照惯例和良好做法，都缩进了这样的界限。在没有花括号和缩进的情况下，原始程序员的意图是完全不清楚的，也许实际上不是要执行的内容。另外，这样的代码很可能会使维护者感到困惑。


**Noncompliant Code Example**

**错误的代码示例**

```
if (condition)  // Noncompliant
doTheThing();

doTheOtherThing();
somethingElseEntirely();

foo();
```

**Compliant Solution**

**正确的代码示例**


```
if (condition)
  doTheThing();

doTheOtherThing();
somethingElseEntirely();

foo();
```