### 3."==" and "!=" should not be used when "equals" is overridden

It is equivalent to use the equality == operator and the equals method to compare two objects if the equals method inherited from Object has not been overridden. In this case both checks compare the object references.

But as soon as equals is overridden, two objects not having the same reference but having the same value can be equal. This rule spots suspicious uses of == and != operators on objects whose equals methods are overridden.

如果继承自对象的 equals method没有被重写，则使用等号"==操作符和equals方法来比较两个对象是等价的，否则不能用等号操作符.


**Noncompliant Code Example**

**错误的代码示例**

```
String firstName = getFirstName(); // String overrides equals
String lastName = getLastName();

if (firstName == lastName) { ... }; // Non-compliant; false even if the strings have the same value
```

**Compliant Solution**

**正确的代码示例**

```
String firstName = getFirstName();
String lastName = getLastName();

if (firstName != null && firstName.equals(lastName)) { ... };
```

