### 4."@CheckForNull" or "@Nullable" should not be used on primitive types

"@CheckForNull" 或者 "@Nullable" 注解不应该用于基本类型（byte, short, int, long, float, double, boolean, char）.

By definition, primitive types are not Objects and so they can't be`null`. Adding`@CheckForNull`or`@Nullable`on primitive types adds confusion and is useless.

This rule raises an issue when`@CheckForNull`or`@Nullable`is set on a method returning a primitive type: byte, short, int, long, float, double, boolean, char.

根据定义，原始类型不是对象，因此它们不能为空。 在基本类型上添加``@CheckForNullor``和`` @Nullable``会增加混乱，并且没有用。

当在返回基础类型的方法上设``@CheckForNullor``和`` @Nullable``时，此规则这些基础类型引起问题：字节，短整数，整型，长整数，浮点型，双精度型，布尔型，字符型。


**Noncompliant Code Example**

**错误的代码示例**

```
@CheckForNull
boolean isFoo() {
 ...
}
```

**Compliant Solution**

**正确的代码示例**

```
boolean isFoo() {
 ...
}
```

