
### 42."File.createTempFile" should not be used to create a directory

File.createTempFile 不够安全，不应该用于创建新的目录


Using`File.createTempFile`as the first step in creating a temporary directory causes a race condition and is inherently unreliable and insecure. Instead,`Files.createTempDirectory`(Java 7+) or a library function such as Guava's similarly-named`Files.createTempDir`should be used.

This rule raises an issue when the following steps are taken in immediate sequence:

-   call to`File.createTempFile`
-   delete resulting file
-   call`mkdir`on the File object

**Note**that this rule is automatically disabled when the project's`sonar.java.source`is lower than`7`.



**Noncompliant Code Example**

**错误的代码示例**

```
File tempDir;
tempDir = File.createTempFile("", ".");
tempDir.delete();
tempDir.mkdir();  // Noncompliant
```


**Compliant Solution**

**正确的代码示例**

```
Path tempPath = Files.createTempDirectory("");
File tempDir = tempPath.toFile();
```


