
### 45."for" loop increment clauses should modify the loops' counters

for 循环应该在增量子句中改循环的计数器


It can be extremely confusing when a`for`loop's counter is incremented outside of its increment clause. In such cases, the increment should be moved to the loop's increment clause if at all possible.

当for循环的计数器在其递增子句之外递增时，可能会造成极大的混乱。在这种情况下，如果可能的话，应将增量移到循环的增量子句中。



**Noncompliant Code Example**

**错误的代码示例**

```
for (i = 0; i < 10; j++) { // Noncompliant
  // ...
  i++;
}
```


**Compliant Solution**

**正确的代码示例**

```
for (i = 0; i < 10; i++, j++) {
  // ...
}
```
或者


```
for (i = 0; i < 10; i++) {
  // ...
  j++;
}
```


