
### 46."for" loop stop conditions should be invariant

循环停止的条件应该是不可变的

A`for`loop stop condition should test the loop counter against an invariant value (i.e. one that is true at both the beginning and ending of every loop iteration). Ideally, this means that the stop condition is set to a local variable just before the loop begins.

for循环停止条件应针对不变值测试循环计数器（即在每次循环迭代的开始和结束时都为真的值）。 理想情况下，这意味着将在循环开始之前将停止条件设置为局部变量。


Stop conditions that are not invariant are slightly less efficient, as well as being difficult to understand and maintain, and likely lead to the introduction of errors in the future.

不变的停止条件效率稍低，并且难以理解和维护，并且可能导致将来引入错误。

This rule tracks three types of non-invariant stop conditions:

-   When the loop counters are updated in the body of the`for`loop
-   When the stop condition depend upon a method call
-   When the stop condition depends on an object property, since such properties could change during the execution of the loop.

此规则跟踪三种类型的非停止条件：

- 在for循环的主体中更新循环计数器时
- 当停止条件取决于方法调用时
- 当停止条件取决于对象属性时，由于这些属性在循环执行期间可能会更改。



**Noncompliant Code Example**

**错误的代码示例**

```
for (int i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant; counter updated in the body of the loop
  ...
}
```


**Compliant Solution**

**正确的代码示例**

```
for (int i = 0; i < 10; i++) {...}
```


