### 47."getClass" should not be used for synchronization

getClass方法不应该被用在synchronization 修饰的方法中

`getClass`should not be used for synchronization in non-`final`classes because child classes will synchronize on a different object than the parent or each other, allowing multiple threads into the code block at once, despite the`synchronized`keyword.

Instead, hard code the name of the class on which to synchronize or make the class`final`.

getClass不应在非最终类中用于同步，因为子类将在与父类或彼此之间不同的对象上进行同步，尽管使用了synced关键字，但允许一次将多个线程插入代码块。

相反，硬编码要在其上进行同步或使其最终化的类的名称。


**Noncompliant Code Example**

**错误的代码示例**

```
public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (this.getClass()) {  // Noncompliant
      // ...
    }
  }
```


**Compliant Solution**

**正确的代码示例**

```
public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (MyClass.class) {
      // ...
    }
  }
```


