### 100."Stream.peek" should be used with caution

.peek()方法主要用于debug，而且仅在对流内元素进行操作时，peek()才会被调用，所以在使用时需要谨慎

According to its JavaDocs,`java.util.Stream.peek()`“exists mainly to support debugging” purposes. Although this does not mean that using it for other purposes is discouraged, relying on`peek()`without careful consideration can lead to error-prone code such as:

-   If the stream pipeline does not include a terminal operation, no elements will be consumed and the`peek()`action will not be invoked at all.
-   As long as a stream implementation can reach the final step, it can freely optimize processing by only producing some elements or even none at all (e.g. relying on other collection methods for counting elements). Accordingly, the`peek()`action will be invoked for fewer elements or not at all.

This rule raises an issue for each use of`peek()`to be sure that it is challenged and validated by the team to be meant for production debugging/logging purposes.

**Noncompliant Code Example**

**错误的代码示例**

```
Stream.of("one", "two", "three", "four")
         .filter(e -> e.length() > 3)
         .peek(e -> System.out.println("Filtered value: " + e)); // Noncompliant
```

