### 95."SingleConnectionFactory" instances should be set to "reconnectOnException"

“SingleConnectionFactory”实例应该设置为“reconnectOnException”，因为在不启用reconnectOnException设置的情况下使用SingleConnectionFactory将 在连接出错时无法自动恢复连接

Use of a Spring`SingleConnectionFactory`without enabling the`reconnectOnException`setting will prevent automatic connection recovery when the connection goes bad.

That's because the`reconnectOnException`property defaults to`false`. As a result, even if the code that uses this connection factory (Spring's`DefaultMessageListenerContainer`or your own code) has reconnect logic, that code won't work because the`SingleConnectionFactory`will act like a single-connection pool by preventing connection`close`calls from actually closing anything. As a result, subsequent factory`create`operations will just hand back the original broken`Connection`.

**Noncompliant Code Example**

**错误的代码示例**

```
<bean id="singleCF" class="org.springframework.jms.connection.SingleConnectionFactory">  <!-- Noncompliant -->
   <constructor-arg ref="dummyConnectionFactory" />
 </bean>
```

**Compliant Solution**

**正确的代码示例**


```
 <bean id="singleCF" class="org.springframework.jms.connection.SingleConnectionFactory" p:reconnectOnException="true">
   <constructor-arg ref="dummyConnectionFactory" />
 </bean>
```
或者

```
 <bean id="singleCF" class="org.springframework.jms.connection.SingleConnectionFactory">
   <constructor-arg ref="dummyConnectionFactory" />
   <property name="reconnectOnException"><value>true</value></property>
 </bean>
```
