### 61."Iterator.next()" methods should throw "NoSuchElementException"

Iterator.next() 方法应该抛出 NoSuchElementException 异常

By contract, any implementation of the`java.util.Iterator.next()`method should throw a`NoSuchElementException`exception when the iteration has no more elements. Any other behavior when the iteration is done could lead to unexpected behavior for users of this`Iterator`.

根据约定，当迭代中没有更多元素时，java.util.Iterator.next() 方法的任何实现都应引发NoSuchElementException异常。 迭代完成后，任何其他行为都可能导致此Iterator用户的意外行为。

**Noncompliant Code Example**

**错误的代码示例**

```
public class MyIterator implements Iterator<String>{
  ...
  public String next(){
    if(!hasNext()){
      return null;
    }
    ...
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
public class MyIterator implements Iterator<String>{
  ...
  public String next(){
    if(!hasNext()){
      throw new NoSuchElementException();
    }
    ...
  }
}
```


