### 62."java.lang.Error" should not be extended

java.lang.Error 不应该被继承

`java.lang.Error`and its subclasses represent abnormal conditions, such as`OutOfMemoryError`, which should only be encountered by the Java Virtual Machine.


**Noncompliant Code Example**

**错误的代码示例**

```
public class MyException extends Error { /* ... */ }       // Noncompliant
```

**Compliant Solution**

**正确的代码示例**


```
public class MyException extends Exception { /* ... */ }   // Compliant
```