### 63."java.nio.Files#delete" should be preferred

因为java.io.Files#delete会抛异常并说明异常的原因，所以在删除文件时应优先考虑java.nio.Files#delete方法

When`java.io.File#delete`fails, this`boolean`method simply returns`false`with no indication of the cause. On the other hand, when`java.nio.Files#delete`fails, this`void`method returns one of a series of exception types to better indicate the cause of the failure. And since more information is generally better in a debugging situation,`java.nio.Files#delete`is the preferred option.


**Noncompliant Code Example**

**错误的代码示例**

```
public void cleanUp(Path path) {
  File file = new File(path);
  if (!file.delete()) {  // Noncompliant
    //...
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
public void cleanUp(Path path) throws NoSuchFileException, DirectoryNotEmptyException, IOException{
  Files.delete(path);
}
```