### 65."javax.crypto.NullCipher" should not be used for anything other than testing

javax.crypto.NullCipher 应该只用于测试

By contract, the`NullCipher`class provides an "identity cipher"one that does not transform or encrypt the plaintext in any way. As a consequence, the ciphertext is identical to the plaintext. So this class should be used for testing, and never in production code.


**Noncompliant Code Example**

**错误的代码示例**

```
NullCipher nc = new NullCipher();
```
