### 66."Lock" objects should not be "synchronized"

“锁定”对象不应使用“synchronized”

`java.util.concurrent.locks.Lock`offers far more powerful and flexible locking operations than are available with`synchronized`blocks. So synchronizing on a`Lock`throws away the power of the object, and is just silly. Instead, such objects should be locked and unlocked using`tryLock()`and`unlock()`.

与同步块相比，java.util.concurrent.locks.Lock提供了更强大，更灵活的锁定操作。 因此，同步上锁会丢失对象的功能，这很愚蠢。 而是应使用 tryLock()和 unLock()  锁定和解锁此类对象。

**Noncompliant Code Example**

**错误的代码示例**

```
Lock lock = new MyLockImpl();
synchronized(lock) {  // Noncompliant
  //...
}
```

**Compliant Solution**

**正确的代码示例**


```
Lock lock = new MyLockImpl();
lock.tryLock();
//...
```

