### 68."Map.get" and value test should be replaced with single method call

需要同时获取map的键值对时，应该使用单个的获取方法

It's a common pattern to test the result of a`java.util.Map.get()`against`null`before proceeding with adding or changing the value in the map. However the`java.util.Map`API offers a significantly better alternative in the form of the`computeIfPresent()`and`computeIfAbsent()`methods. Using these instead leads to cleaner and more readable code.

**Note**that this rule is automatically disabled when the project's`sonar.java.source`is not 8.

在继续添加或更改映射中的值之前，通常将java.util.Map.get（）的结果测试为null。 但是，java.util.Map API以computeIfPresent（）和computeIfAbsent（）方法的形式提供了一种更好的替代方法。 相反，使用这些代码将导致代码更简洁，更具可读性。 请注意，当项目的sonar.java.source不是8时，将自动禁用此规则。


**Noncompliant Code Example**

**错误的代码示例**

```
V value = map.get(key);
if (value == null) {  // Noncompliant
  value = V.createFor(key);
  if (value != null) {
    map.put(key, value);
  }
}
return value;
```

**Compliant Solution**

**正确的代码示例**


```
return map.computeIfAbsent(key, k -> V.createFor(k));
```