### 69."notifyAll" should be used

应该使用notifyAll方法唤醒所有休眠的进程

`notify`and`notifyAll`both wake up sleeping threads, but`notify`only rouses one, while`notifyAll`rouses all of them. Since`notify`might not wake up the right thread,`notifyAll`should be used instead.


**Noncompliant Code Example**

**错误的代码示例**

```
class MyThread extends Thread{

  @Override
  public void run(){
    synchronized(this){
      // ...
      notify();  // Noncompliant
    }
  }
}
```

**Compliant Solution**

**正确的代码示例**


```
class MyThread extends Thread{

  @Override
  public void run(){
    synchronized(this){
      // ...
      notifyAll();
    }
  }
}
```